ic-auction/src/types.rs

```diff
@@ -0,0 +1,40 @@
+use ic_cdk::export::candid::{CandidType, Deserialize};
+use ic_helpers::tokens::Tokens128;
+
+pub type Cycles = u64;
+pub type TxId = u64;
+pub type Timestamp = u64;
+
+#[derive(CandidType, Debug, Clone, Deserialize, PartialEq)]
+pub struct AuctionInfo {
+    pub auction_id: usize,
+    pub auction_time: Timestamp,
+    pub tokens_distributed: Tokens128,
+    pub cycles_collected: Cycles,
+    pub fee_ratio: f64,
+    pub first_transaction_id: TxId,
+    pub last_transaction_id: TxId,
+}
+
+/// Current information about upcoming auction and current cycle bids.
+#[derive(CandidType, Debug, Clone, Deserialize)]
+pub struct BiddingInfo {
```

Comment:
Same question as the above struct (does it need to be public)

Reply:
I exported it for the is20 to overwrite the endpoints to follow is20 spec

Reply:
yeah if they are needed outside of the crate then all good